platform.txt

```diff
@@ -85,6 +85,16 @@ compiler.size.cmd=xtensa-lx106-elf-size
 # This can be overridden in boards.txt
 build.extra_flags=-DESP8266
 
+# Pre and post build hooks
+build.opt.name=build_opt.h
```

Comment:
`build_opt.h` is not going to be a valid C or C++ header.  Can we use a different extension and call it something like `build.opt` just to make it blindingly obvious?

Reply:
I've tried to use something else, but only .h files have been successful. I guess this is connected on how Arduino detects files, but I'm not sure

Reply:
Ah, good point.  The Arduino IDE won't copy every file to INO directory the build dirs.  I remember hitting this with some other things a while back.